code/go/0chain.net/blobbercore/reference/object.go

```diff
@@ -3,77 +3,48 @@ package reference
 import (
 	"context"
 	"path/filepath"
-	"strings"
 
 	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/datastore"
 	"github.com/0chain/blobber/code/go/0chain.net/core/common"
-	_ "gorm.io/gorm"
-	"gorm.io/gorm/clause"
 )
 
-// DeleteObject delete object from tree, and return tree root and deleted content hash list
-func DeleteObject(ctx context.Context, allocationID, path string) (*Ref, map[string]bool, error) {
-	rootRef, err := GetObjectTree(ctx, allocationID, "/")
-	if err != nil {
-		return nil, nil, err
-	}
-
-	rootRef.HashToBeComputed = true
-	db := datastore.GetStore().
-		GetTransaction(ctx)
+func DeleteObject(ctx context.Context, allocationID, objPath string) (*Ref, error) {
 
-	var deletedObjects []*Ref
-	txDelete := db.Clauses(clause.Returning{Columns: []clause.Column{{Name: "content_hash"}, {Name: "thumbnail_hash"}, {Name: "type"}}})
+	db := datastore.GetStore().GetTransaction(ctx)
+	err := db.Delete(&Ref{}, "allocation_id=? AND path LIKE ? AND path != ?",
+		allocationID, objPath+"%", "/").Error
 
-	path = filepath.Join("/", path)
-	txDelete = txDelete.Where("allocation_id = ? and deleted_at IS NULL and (path LIKE ? or path = ?) and path != ? ", allocationID, path+"%", path, "/")
-
-	err = txDelete.Delete(&deletedObjects).Error
 	if err != nil {
-		return nil, nil, common.NewError("bad_db_operation", err.Error())
+		return nil, err
 	}
 
-	deletedFiles := make(map[string]bool)
-	for _, it := range deletedObjects {
-		if it.Type == FILE {
-			deletedFiles[it.ContentHash] = true
-			if it.ThumbnailHash != "" {
-				deletedFiles[it.ThumbnailHash] = true
-			}
-		}
-	}
-
-	// remove deleted object from tree
-	if path == "/" {
-		rootRef.Children = nil
-		return rootRef, deletedFiles, nil
+	parentPath := filepath.Dir(objPath)
+	rootRef, err := GetReferencePath(ctx, allocationID, parentPath)
+	if err != nil {
+		return nil, err
 	}
 
-	path = strings.TrimSuffix(path, "/")
-	tSubDirs := GetSubDirsFromPath(path)
+	subDirs := GetSubDirsFromPath(parentPath)
 	dirRef := rootRef
-	treelevel := 0
-	for treelevel < len(tSubDirs)-1 {
-		found := false
-		for _, child := range dirRef.Children {
-			if child.Name == tSubDirs[treelevel] && child.Type == DIRECTORY {
-				dirRef = child
-				dirRef.HashToBeComputed = true
+
+	for _, subDir := range subDirs {
+		var found bool
+		for _, ref := range dirRef.Children {
+			if ref.Name == subDir && ref.Type == DIRECTORY {
```

Comment:
Perhaps we should break immediately if the dirRef.Children do not have the subDir, basically this checking all start from the root path, if any level of the path do not exist in the corresponding dirRef, there must be something wrong.

Reply:
It does break with `return` statement.